compiler/zrc_typeck/src/typeck/declaration.rs

```diff
@@ -346,6 +346,15 @@ pub fn process_declaration<'input>(
 
             Some(TypedDeclaration::GlobalLetDeclaration(typed_declarations))
         }
+        AstDeclaration::ModuleAsm { assembly } => {
+            // Type check the assembly expression (should be a string literal)
+            let scope = global_scope.create_subscope();
```

Comment:
Does module assembly truly need to live in a subscope? What kind of things can it clobber the glob-scope in?